modules/storage-api/src/test/java/org/apache/ignite/internal/storage/DataStorageManagerTest.java

```diff
@@ -45,27 +54,85 @@
     void testStorageEngineDuplicate() {
         String sameName = UUID.randomUUID().toString();
 
-        StorageException exception = assertThrows(
-                StorageException.class,
-                () -> new DataStorageManager(Mockito.mock(ConfigurationRegistry.class), workDir) {
-                    /** {@inheritDoc} */
-                    @Override
-                    protected Iterable<StorageEngineFactory> engineFactories() {
-                        return List.of(
-                                (configRegistry, storagePath) -> createMockedStorageEngine(sameName),
-                                (configRegistry, storagePath) -> createMockedStorageEngine(sameName)
-                        );
-                    }
-                });
+        IllegalStateException exception = assertThrows(
+                IllegalStateException.class,
+                () -> new DataStorageManager(
+                        mock(ConfigurationRegistry.class),
+                        workDir,
+                        List.of(
+                                createMockedStorageEngineFactory(sameName),
+                                createMockedStorageEngineFactory(sameName)
+                        )
+                ));
 
         assertThat(exception.getMessage(), Matchers.startsWith("Duplicate key"));
     }
 
-    private StorageEngine createMockedStorageEngine(String name) {
-        StorageEngine mocked = Mockito.mock(StorageEngine.class);
+    @Test
+    void testDefaultTableDataStorageConsumerSingleStorageEngine() {
+        String engineName = UUID.randomUUID().toString();
+
+        DataStorageManager dataStorageManager = new DataStorageManager(
+                mock(ConfigurationRegistry.class),
+                workDir,
+                List.of(createMockedStorageEngineFactory(engineName))
+        );
+
+        checkDefaultTableDataStorageConsumer(dataStorageManager, engineName, engineName);
+
+        checkDefaultTableDataStorageConsumer(dataStorageManager, UNKNOWN_DATA_STORAGE, engineName);
+    }
+
+    @Test
+    void testDefaultTableDataStorageConsumerMultipleStorageEngines() {
+        String engineName1 = UUID.randomUUID().toString();
+        String engineName2 = UUID.randomUUID().toString();
+
+        DataStorageManager dataStorageManager = new DataStorageManager(
+                mock(ConfigurationRegistry.class),
+                workDir,
+                List.of(
+                        createMockedStorageEngineFactory(engineName1),
+                        createMockedStorageEngineFactory(engineName2)
+                )
+        );
+
+        checkDefaultTableDataStorageConsumer(dataStorageManager, engineName1, engineName1);
+
+        checkDefaultTableDataStorageConsumer(dataStorageManager, engineName2, engineName2);
+
+        checkDefaultTableDataStorageConsumer(dataStorageManager, UNKNOWN_DATA_STORAGE, null);
+    }
+
+    private static void checkDefaultTableDataStorageConsumer(
```

Comment:
It would be nice to have a comment - what do we check here? Mockito-based tests are notoriously hard to read

Reply:
Added a description.